pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -693,6 +693,26 @@ func addRoleConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
 		config["disable-etcd"] = true
 	}
 
+	if isControlPlane(machine) {
+		certArg := fmt.Sprintf("cert-dir=/var/lib/rancher/%s/server/tls/kube-controller-manager", rancherruntime.GetRuntime(controlPlane.Spec.KubernetesVersion))
+		securePortArg := "secure-port=10257"
+		if v, ok := config["kube-controller-manager-arg"]; ok {
+			switch v.(type) {
+			case []string:
+				config["kube-controller-manager-arg"] = append(v.([]string), certArg, securePortArg)
+			case string:
+				// If the argument that is being overridden is either the cert dir or the secure port, we override it
+				if arg := strings.SplitN(v.(string), "=", 2); len(arg) == 2 && (arg[0] == "cert-dir" || arg[0] == "secure-port") {
+					config["kube-controller-manager-arg"] = []string{certArg, securePortArg}
```

Comment:
I think it would be better to figure out how to incorporate a user-defined setting into the probe than it is to override the user setting.